### ISSUE TITLE
Transform audio track component on search results into composite element for quicker navigation

### ISSUE THREAD

**Issue by sarayourfriend** [WordPress/openverse-frontend#1297]
## Problem

Each audio result has three tabbable elements on the single audio view. Takes a very long time to tab through the audio results.

## Description

The audio track component used for search results should be turned into a composite element with special navigation for entering and exiting the composite element.

See [WAI-ARIA grid](https://www.w3.org/TR/wai-aria-1.1/#grid) specification for more details about how to implement this accessibly.

## Additional context

This can be ignored for the all-content (boxed) audio track as the problem is already solved there.

## Design solution
- Design introduction: [Presentation](https://www.figma.com/proto/GeG5nSTdPI887CH39jzjzo/Audio-track-navigation?page-id=26%3A13657&node-id=37%3A28663&scaling=min-zoom&starting-point-node-id=37%3A28663)
- Figma file: [Audio track navigation](https://www.figma.com/file/GeG5nSTdPI887CH39jzjzo/Audio-track-navigation?node-id=0%3A1)
- Design issue #1549

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

**Comment by sarayourfriend** [comment]
For this issue I think rather than implement the composite role (which has tons of edge cases and complexity in our particular case because we have elements that need to be interacted with using the keyboard), I think it would make more sense to transform each audio track result row into an "interactive" link similar to the boxed audio is today. In those code you would tab through results individually rather than their controls. Hitting enter on the result would activate the link container for the entire widget. Then the same interactions that exist on the boxed audio would exist for keyboard users on the row layout as well.

We run into a problem here however (which we would have with the composite widget anyway), which is that sighted keyboard users who do not use a screen reader will not have indications about how to interact with the audio widget. I think we'd need to write some "keyboard user detection" script that would show a tooltip (or some other non-jumpy UI solution) to indicate to keyboard users that they can interact with the audio widget. I think we actually have this problem today on the all content results as well.

@panchovm do you have any thoughts about how best to implement this pattern? Also while I can take an educated guess about how best to add the focus ring to the audio row layout, it'd be nice to get an idea from you as well :slightly_smiling_face:

**Comment by sarayourfriend** [comment]
Another thought about this, we'll need to make the link element have a slider role and announce the track time and such. Essentially the entire row layout will behave as though it were a single cohesive audio playback widget. Luckily this code already exists in the waveform, it'll just need to be hoisted up to the container level for the row layout.

**Comment by panchovm** [comment]
The focus ring style for the audio track is already in [the design library](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=2275%3A4134), that if I understood the wrapping area of the focus area correctly. Here is a screenshot covering from the whole track to the wave area.

<img width="500" alt="CleanShot 2022-06-16 at 14 27 11@2x" src="https://user-images.githubusercontent.com/895819/174140381-cbe591db-e25b-4598-b710-cb24d4550d5f.png">

Regarding the tooltip element, have you seen a best practice case? I am thinking of showing the tooltip out of the focus area in the top-left part, and in the top-right for RTL settings. I will search for examples.

**Comment by sarayourfriend** [comment]
Ah great, I should have looked. We would only be using the "group focus" one now as the rest of the elements would be unfocusable via keyboard and just be interactable using keyboard when the group is focused on using mouse by clicking each discrete element.

A tooltip sounds okay, but it could be long :thinking: And for mobile keyboard users I'm not sure how well it would operate. Here's some of the text that would probably need to be in the tooltip:

> Use the <kbd>Left arrow</kbd>/<kbd>Right arrow</kbd> keys to seek the audio.
> Use the <kbd>Spacebar</kbd> to play/pause the audio.

The arrow keys may be able to be written with unicode symbols (like 🡢) and have aria-labels added if the symbols aren't read in a sensible manner by screen readers.

**Comment by panchovm** [comment]
Now I understand. Thanks.

After tackling other current design issues, I will start working on this problem and share a proposal in a new ticket.

**Comment by sarayourfriend** [comment]
For whoever picks up this issue: I've got a branch with a significant part of the base work done for this: https://github.com/wordpress/openverse-frontend/tree/add/audio-track-composite-layout

There are failing tests, but the bulk of what it needs is:

* complete the work on the `useSeekable` composable to make the `VAudioTrack` itself respond to keyboard seek interactions
* fix the design
* fix click interactions (need to prevent click events from bubbling on the waveform and the play/pause button)

**Comment by panchovm** [comment]
I have updated the initial message with the proposal and discussion ticket (#1549) and the Figma file link.
